File: server/Models/StockPhoto.cs
namespace Server.Models;

/// <summary>A photo of a <see cref="Stock"/>, tagged with the date it was actually taken (which
/// may differ from <see cref="CreatedAt"/>, the upload time).</summary>
public class StockPhoto
{
    public int Id { get; set; }
    public int StockId { get; set; }
    public string ImagePath { get; set; } = string.Empty;
    public DateOnly TakenAt { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}

File: web/src/types/stock-photo.ts
export type StockPhoto = {
  id: number;
  stockId: number;
  imagePath: string;
  /** Date the photo was actually taken (`YYYY-MM-DD`), not the upload time. */
  takenAt: string;
  createdAt: string;
};

export type StockPhotoInput = {
  stockId: number;
  imagePath: string;
  takenAt: string;
};
